.gitlab/ci/global.yml

```diff
@@ -13,66 +13,235 @@
-    - mkdir -p -m 777 $ARTIFACTS_FOLDER/
-    - |
-      if [[ -f "$BASH_ENV" ]]; then
-        source "$BASH_ENV"
-      fi
-    - source .circleci/content_release_vars.sh
-    - export PATH="${PWD}/node_modules/.bin:${PATH}"
+.prepare_env: &prepare_env
+  - source .gitlab/helper_functions.sh
+  - echo "=== Running before script ==="
+  - git checkout $CI_COMMIT_BRANCH
+  - git config diff.renameLimit 6000
+  - mkdir -p -m 777 $ARTIFACTS_FOLDER/
+  - |
+    if [[ -f "$BASH_ENV" ]]; then
+      source "$BASH_ENV"
+    fi
+  - source .circleci/content_release_vars.sh
+  - export PATH="${PWD}/node_modules/.bin:${PATH}"
+
+  - section_start "Creating new clean logs folder" --collapsed
+  - rm -rf $ARTIFACTS_FOLDER/logs
+  - mkdir -p $ARTIFACTS_FOLDER/logs
+  - section_end "Creating new clean logs folder"
+
+  - section_start "Granting execute permissions on files" --collapsed
+  - chmod +x ./Tests/scripts/*
+  - chmod +x ./Tests/Marketplace/*
+  - section_end "Granting execute permissions on files"
 
-    - section_start "Creating new clean logs folder" --collapsed
-    - rm -rf $ARTIFACTS_FOLDER/logs
-    - mkdir -p $ARTIFACTS_FOLDER/logs
-    - section_end "Creating new clean logs folder"
+  - section_start "Installing Virtualenv" --collapsed
+  - |
+    if [ -f "./venv/bin/activate" ]; then
+      echo "found venv"
+    else
+      echo "installing venv"
+      NO_HOOKS=1 SETUP_PY2=yes .hooks/bootstrap >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+    fi
+  - venv/bin/pip3 install -r .circleci/build-requirements.txt >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+  - source ./venv/bin/activate
+  - |
+    if [ -n "${DEMISTO_SDK_NIGHTLY}" ]; then
+      echo "Installing SDK from master branch" | tee --append $ARTIFACTS_FOLDER/logs/installations.log
+      pip3 install git+https://github.com/demisto/demisto-sdk@master >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+    fi
+  - section_end "Installing Virtualenv"
 
-    - section_start "Granting execute permissions on files" --collapsed
-    - chmod +x ./Tests/scripts/*
-    - chmod +x ./Tests/Marketplace/*
-    - section_end "Granting execute permissions on files"
+  - section_start "Installing SSH keys" --collapsed
+  - eval $(ssh-agent -s)
+  - chmod 400 $OREGON_CI_KEY
+  - ssh-add $OREGON_CI_KEY
+  - mkdir -p ~/.ssh
+  - chmod 700 ~/.ssh
+  - section_end "Installing SSH keys"
 
-    - section_start "Installing Virtualenv" --collapsed
+  - section_start "Build Parameters"
+  - set | grep -E "^NIGHTLY=|^INSTANCE_TESTS=|^SERVER_BRANCH_NAME=|^ARTIFACT_BUILD_NUM=|^DEMISTO_SDK_NIGHTLY=|^TIME_TO_LIVE=|^CONTRIB_BRANCH=|^FORCE_PACK_UPLOAD=|^PACKS_TO_UPLOAD=|^BUCKET_UPLOAD=|^GCS_MARKET_BUCKET=|^SLACK_CHANNEL="
+  - python --version
+  - python3 --version
+  - node --version
+  - npm --version
+  - demisto-sdk --version
+  - section_end "Build Parameters"
+
+.default-before_script:
+  before_script: *prepare_env
+
+.default-after-script:
+  after_script:
+    - *prepare_env
```

Comment:
Do we need everything in the `prepare_env` configuration anchor in order to properly run the slack notification and instance deletion?

Reply:
yup, it fixes the connection errors as well. after_scripts run on a different bash so we need to install stuff

Reply:
The after_script runs on a different bash so we kinda need to prepare the env all over again, plus this fixed the issues with the destroy_instances problems